src/Stream.php

```diff
@@ -222,7 +222,9 @@ public function isWritable()
         }
 
         $meta = stream_get_meta_data($this->resource);
-        return is_writable($meta['uri']);
+        $mode = $meta['mode'];
+
+        return (strstr($mode, 'w') || strstr($mode, '+'));
```

Comment:
Answer is complex, depending on your definition  of "valid". If by "valid" you mean everything that PHP accepts, then:
* for TEMP and MEMORY streams, `mode` is normalized, you always get `rb` or `w+b`
* for files PHP will accept even things like this: `$f2 = fopen('/tmp/test', 'ah()+');`, and report them under `mode`

So, apparently in order to satisfy `fopen` docs, we have to check if any of characters `+, a, c, w, x` is present in `mode`. If you agree, I can post fix in a moment.

Yes, `w` can be before `+`. Not sure why you ask though.